src/rgw/store/dbstore/common/dbstore.cc

```diff
@@ -54,11 +54,29 @@ int DB::Initialize(string logfile, int loglevel)
   return ret;
 }
 
+int DB::createGC(const DoutPrefixProvider *dpp) {
+  int ret = 0;
+  /* create gc thread */
+
+  gc_worker = std::make_unique<DB::GC>(dpp, this);
```

Comment:
I wonder about the general approach of introducing a singleton thread to do cleanup, but don't really object either;  it will eventually run into some issues, and very possibly we will wish we were using asio/async primitives in the dbstore as well as rgw;  unfortunately, we don't really have full examples of that working, and maybe even if we want that, it would be better to fold it in later

Reply:
ACK. Apart from GC I think even this objectID/ handling I/O concurrency probably may need generic approach in future.

In the current rgw_op implementation all the reads or writes are not atomic but include multiple steps ( - prepare, write/iterate_data, complete). So all the backend stores need to handle this issue and most likely will use some uniqueID (atleast in the case of both dbstore & motr) to differentiate the uploads of the same object. So maybe this identifier needs to be part of sal::Object itself.